### ISSUE TITLE
BUG: Unable to install pytensor as a package with pip install -e

### ISSUE THREAD

**Issue by tanish1729** [pymc-devs/pytensor#947]
### Describe the issue:

I installed pytensor using `pip install -e <path_to_my_fork>` but i keep running into errors after that and cant even do simple imports. Everything seems to be failing because of 
![image](https://github.com/user-attachments/assets/5b30f56c-7d82-4c0c-92ea-045e37f6b97e)

### Reproducable code example:

```python
import pytensor
print(pytensor.__version__)
```

### Error message:

```shell
AttributeError: module 'pytensor' has no attribute '__version__'
```

### PyTensor version information:

Can't check version

**Comment by jessegrabowski** [comment]
cc: @Armavica @maresb

**Comment by maresb** [comment]
Hi @tanish1729, this is concerning, thanks so much for reporting it!

Let's try and diagnose this.

Please try running

```python
import pytensor

print(pytensor.__file__)
```

This should report the location of `__init__.py` where `__version__` is defined. I'd first ensure that the Python interpreter you're running is finding the expected copy of PyTensor.

The next thing I'd look at is if your version of `__init__.py` is up-to-date. (@Armavica made some changes last week, so perhaps the issue is that you need to pull in those changes.) The `__version__ attribute is defined on [this line](https://github.com/pymc-devs/pytensor/blob/ad27dc752798dfb75ed8d1c0a9b840859a8a898d/pytensor/__init__.py#L34). If that line executes, then I wouldn't expect the missing attribute you're seeing.

I hope it's something simple like this and not something more serious.

**Comment by tanish1729** [comment]
<img width="842" alt="image" src="https://github.com/user-attachments/assets/50817dc8-d794-4440-af04-f74d8692adf3">

is pytensor not installed or something? cuz it is showing me the location when i try to just see pytensor

**Comment by maresb** [comment]
Hi @tanish1729, thanks for the info!

I suspect that Python could be getting confused with a conflicting directory named `pytensor`.

What's your current directory and Python search path? You can find them with

```python
from pathlib import Path
import sys

print(Path.cwd())
print(sys.path)
```

**Comment by tanish1729** [comment]
<img width="989" alt="image" src="https://github.com/user-attachments/assets/de3af606-1845-4bf3-8a59-298c42ce0fba">
i already uninstalled pytensor and did `conda clean` and the new package always gets installed with the name its showing there

**Comment by maresb** [comment]
I have a theory on what's going wrong.

First I assume that your clone of PyTensor is in `/Users/tanish/Desktop/idk/pymc-gsoc/pytensor`.

If you are running VS Code / Jupyter from `/Users/tanish/Desktop/idk/pymc-gsoc` and import `pytensor` then it will see `/Users/tanish/Desktop/idk/pymc-gsoc/pytensor` as the Python project because the directory is named `pytensor`. Instead, it needs to find the directory `/Users/tanish/Desktop/idk/pymc-gsoc/pytensor/pytensor` (note `pytensor` twice at the end).

If you're in VS Code, then you should open the folder `/Users/tanish/Desktop/idk/pymc-gsoc/pytensor` instead of `/Users/tanish/Desktop/idk/pymc-gsoc`. Similarly, if you're running Jupyter, you should start it inside `/Users/tanish/Desktop/idk/pymc-gsoc/pytensor`.

I find this behavior from Python extremely frustrating.

I'm very tempted to suggest the following, but I'm not sure if this potentially introduces new issues:

Put the following contents into `/Users/tanish/Desktop/idk/pymc-gsoc/pytensor/__init__.py` and try what you were doing before.

```python
from pathlib import Path

likely_repo_root = Path(__file__).parent

raise RuntimeError(
    f"Python is looking for PyTensor in {likely_repo_root}, but it's "
    f"actually located in {likely_repo_root / 'pytensor'}. Probably "
    f"you need to change your working directory from {Path.cwd()} "
    f"to {likely_repo_root}."
)
```

**Comment by maresb** [comment]
Turned this into a PR in #949

**Comment by tanish1729** [comment]
hey @maresb, yes it did start working fine after i went into the pytensor directory containing all the files.
however, this is kinda redundant for what i wanted to do. if im already inside the cloned directory and running all the imports, isnt that the same as using the package locally (i.e I should be able to do this without pip install as well).
what i wanted to do was install the current version of my local pytensor on the system so it could run in other places too.

**Comment by maresb** [comment]
> if im already inside the cloned directory and running all the imports, isnt that the same as using the package locally

Yes, but in this case you're clobbering the `pip install`ed PyTensor with the PyTensor from your current directory.

As long as you've `pip install`ed it, it should work anywhere else that doesn't have a conflicting `pytensor` subdirectory. Does this make sense?

**Comment by tanish1729** [comment]
yes i tried that and it was working. i get what the problem is now. thanks!
